File: record/Google_Cloud_Speech_api/wav_text.py
import os
import io
from google.cloud import speech
os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = 'hogehoge.json'


# Instantiates a client
client = speech.SpeechClient()
### 音声データを指定
speech_file = '../wav/hello.wav'

### rb(read binary)でデータを読み込む
with io.open(speech_file, 'rb') as f:
    content = f.read()

### RecognitionAudioにデータを渡す
audio = speech.RecognitionAudio(content=content)

config = speech.RecognitionConfig(
    ### encodeでエラーが出たのでENCODING_UNSPECIFIEDに変更
    encoding=speech.RecognitionConfig.AudioEncoding.ENCODING_UNSPECIFIED,
    sample_rate_hertz=16000,
    language_code="ja-JP",
)

### 音声を抽出
response = client.recognize(config=config, audio=audio)

### 抽出結果をprintで表示
for result in response.results:
    print("Transcript: {}".format(result.alternatives[0].transcript))

File: record/Google_Cloud_Speech_api/wav_text.php
<?php

// オーディオ ファイルを読み込む
$filename = 'hello.wav';
$fp = fopen( $filename, 'rb' );
$content = fread( $fp, filesize( $filename ) );
fclose( $fp );

// それをBase64エンコードする
$encodedContent = base64_encode( $content );

// JSON形式でリクエスト ボディを生成する
$requestBody = <<<EOM
{
  "config": {
    "encoding": "FLAC",
    "sampleRateHertz": 48000,
    "languageCode": "ja-JP"
  },
  "audio": {
    "content": "{$encodedContent}"
  }
}
EOM;

$apiKey = ''; // APIキー
$url = 'https://speech.googleapis.com/v1/speech:recognize?key='.$apiKey;

$header = array( 'Content-Type: application/json; charset=utf-8' );

$ch = curl_init( $url );
curl_setopt( $ch, CURLOPT_HTTPHEADER,  $header );
curl_setopt( $ch, CURLOPT_POSTFIELDS, $requestBody );
curl_setopt( $ch, CURLOPT_FAILONERROR, TRUE );
curl_setopt( $ch, CURLOPT_SSL_VERIFYPEER, FALSE );

// リクエストし、結果を出力する
if (curl_exec($ch) === FALSE) echo curl_error($ch);
curl_close($ch);
